src/pages/OAuth2Page.tsx

```diff
@@ -13,22 +16,29 @@ const OAuth2Page: React.FC = () => {
   const history = useHistory()
   const { isAuthenticating, currentMemberId, socialLogin } = useAuth()
   const updateYoutubeChannelIds = useUpdateMemberYouTubeChannelIds()
+  const { settings } = useApp()
+  const [code] = useQueryParam('code', StringParam)
+  const [state] = useQueryParam('state', StringParam)
 
   const params = new URLSearchParams('?' + window.location.hash.replace('#', ''))
   const accessToken = params.get('access_token')
-  const state: {
-    provider: string
+
+  const {
+    provider,
+    redirect,
+  }: {
+    provider: 'facebook' | 'google' | 'line'
     redirect: string
-  } = JSON.parse(params.get('state') || '{}')
+  } = JSON.parse(atob(state || ''))
```

Comment:
`JSON.parse` will throw error on empty string
wrapped in try-catch scope or keep `JSON.parse('{}')` with `?` in type definition